packages/cactus-plugin-ledger-connector-corda/src/test/typescript/integration/corda-v5-flow.test.ts

```diff
@@ -0,0 +1,350 @@
+      await apiClient.startFlowV1(aliceListsChat);
+      const aliceList2Response = await pollEndpointUntilCompleted(
+        shortHashAlice,
+        "list-2",
+      );
+      expect(aliceList2Response).toBeTruthy();
+
+      //5. Alice checks the history of the chat with Bob
+      const aliceHistoryRequest = {
+        holdingIDShortHash: shortHashAlice,
+        clientRequestId: "get-1",
+        flowClassName:
+          "com.r3.developers.csdetemplate.utxoexample.workflows.GetChatFlow",
+        requestBody: {
+          id: chatWithBobId,
+          numberOfRecords: "4",
+        },
+      };
+      await apiClient.startFlowV1(aliceHistoryRequest);
+
+      const aliceHistoryResponse = await pollEndpointUntilCompleted(
+        shortHashAlice,
+        "get-1",
+      );
+      expect(aliceHistoryResponse).toBeTruthy();
+
+      //6. Alice replies to Bob
+      const aliceReply = {
+        holdingIDShortHash: shortHashAlice,
+        clientRequestId: "update-4",
+        flowClassName:
+          "com.r3.developers.csdetemplate.utxoexample.workflows.UpdateChatFlow",
+        requestBody: {
+          id: chatWithBobId,
+          message: "I am very well thank you",
+        },
+      };
+
+      await apiClient.startFlowV1(aliceReply);
+      const aliceReplyResponse = await pollEndpointUntilCompleted(
+        shortHashAlice,
+        "update-4",
+      );
+      expect(aliceReplyResponse).toBeTruthy();
+
+      //7. Bob gets the chat history
+      const bobHistoryRequest = {
+        holdingIDShortHash: shortHashBob,
+        clientRequestId: "get-2",
+        flowClassName:
+          "com.r3.developers.csdetemplate.utxoexample.workflows.GetChatFlow",
+        requestBody: {
+          id: chatWithBobId,
+          numberOfRecords: "2",
+        },
+      };
+      await apiClient.startFlowV1(bobHistoryRequest);
+
+      const bobHistoryResponse = await pollEndpointUntilCompleted(
+        shortHashBob,
+        "get-2",
+      );
+      expect(bobHistoryResponse).toBeTruthy();
+
+      //8. List Flows Endpoint Test
+      const queryVar: GetFlowV1Request = {
+        holdingIDShortHash: shortHashBob,
+      };
+      const response = await apiClient.listFlowV1(queryVar);
+      expect(response).toBeTruthy();
+    });
+
+    describe("Negative Testing", () => {
+      it("Negative Test, invalid flow class name", async () => {
+        const invalidFlowName = "nonExistentFlow";
+        const request = {
+          holdingIDShortHash: shortHashBob,
+          clientRequestId: "test-1",
+          flowClassName: invalidFlowName,
+          requestBody: {
+            chatName: "Test-1",
+            otherMember: "CN=Bob, OU=Test Dept, O=R3, L=London, C=GB",
+            message: "Testing",
+          },
+        };
+        try {
+          await apiClient.startFlowV1(request);
+        } catch (error) {
+          expect(error).toBeDefined();
+        }
+      });
+    });
+  });
+
+  async function pollEndpointUntilCompleted(
```

Comment:
This method should be the part of the connector class itself and should be called by the endpoint functions themselves.
So if this function waits after an endpoint is called, then in the endpoint function present in the connector class, you can add it just before returning the function response.

Test cases are used to test the functionality of the existing codebase itself. So adding new functions to the test cases, like these, do not justify the testing purpose, because then you are basically saying that `the existing codebase is inadequate in its offerings`.